### PR TITLE
DRAFT: Adds Transport Layer Header Improvements

### PR DESCRIPTION
**Pull request by asketagarwal**
Draft PR for master repo that changes the way SDK lookups header from the INameValueCollection

### CODE REVIEW

File: Microsoft.Azure.Cosmos/src/direct/INameValueCollection.cs [hunk 1]
```diff
@@ -36,4 +36,261 @@ internal interface INameValueCollection : IEnumerable
...
+        string RbacAction { get; }
+        string RbacResource { get; }
+        string ChangeFeedWireFormatVersion { get; }
+        string PopulateByokEncryptionProgress { get; }
+        string UseUserBackgroundBudget { get; }
+        string IncludePhysicalPartitionThroughputInfo { get; }
+        string Version { get; }
+    }
+
+    internal class RequestHeaderWrapperForNameValue : IRequestHeaders
```
Comment (j82w) [inline Microsoft.Azure.Cosmos/src/direct/INameValueCollection.cs:152]: nit: separate files for different classes

File: Microsoft.Azure.Cosmos/src/direct/INameValueCollection.cs [hunk 2]
```diff
@@ -36,4 +36,261 @@ internal interface INameValueCollection : IEnumerable
...
+            TransportSerialization.FillTokenFromHeader(request, WFConstants.BackendHeaders.PartitionResourceFilter, rntbdRequest.partitionResourceFilter, rntbdRequest);
+            TransportSerialization.FillTokenFromHeader(request, WFConstants.BackendHeaders.EnableDynamicRidRangeAllocation, rntbdRequest.enableDynamicRidRangeAllocation, rntbdRequest);
+            TransportSerialization.FillTokenFromHeader(request, WFConstants.BackendHeaders.SchemaOwnerRid, rntbdRequest.schemaOwnerRid, rntbdRequest);
+            TransportSerialization.FillTokenFromHeader(request, WFConstants.BackendHeaders.SchemaHash, rntbdRequest.schemaHash, rntbdRequest);
+            TransportSerialization.FillTokenFromHeader(request, WFConstants.BackendHeaders.SchemaId, rntbdRequest.schemaId, rntbdRequest);
+            TransportSerialization.FillTokenFromHeader(request, HttpConstants.HttpHeaders.IsClientEncrypted, rntbdRequest.isClientEncrypted, rntbdRequest);
+            TransportSerialization.FillTokenFromHeader(request, WFConstants.BackendHeaders.CorrelatedActivityId, rntbdRequest.correlatedActivityId, rntbdRequest);
+     */
+    internal interface IRequestHeaders
```
Comment (j82w) [inline Microsoft.Azure.Cosmos/src/direct/INameValueCollection.cs:80]: nit: IStoreRequestHeaders

File: Microsoft.Azure.Cosmos/src/Headers/StoreRequestNameValueCollection.cs [hunk 3]
```diff
@@ -15,7 +15,7 @@ namespace Microsoft.Azure.Cosmos
     using Microsoft.Azure.Documents;
     using Microsoft.Azure.Documents.Collections;
 
-    internal class StoreRequestNameValueCollection : CosmosMessageHeadersInternal, INameValueCollection
+    internal class StoreRequestNameValueCollection : CosmosMessageHeadersInternal, INameValueCollection, IRequestHeaders
```
Comment (j82w) [inline Microsoft.Azure.Cosmos/src/Headers/StoreRequestNameValueCollection.cs:18]: This is a generated file. This needs to be changed in https://github.com/Azure/azure-cosmos-dotnet-v3/blob/master/Microsoft.Azure.Cosmos/src/Headers/StoreRequestNameValueCollection.tt

File: Microsoft.Azure.Cosmos/src/direct/INameValueCollection.cs (hunk 1 above)
Comment (asketagarwal) [reply Microsoft.Azure.Cosmos/src/direct/INameValueCollection.cs:152]: yeah, will do that. This is a draft to be ported to the master repo

**Comment by j82w** [comment]
Moved: https://github.com/Azure/azure-cosmos-dotnet-v3/pull/3177

**Pull request closed without merging**